bin/sidecli.re

```diff
@@ -323,6 +323,45 @@ let produce_block = {
   Term.(lwt_ret(const(produce_block) $ key_wallet $ state_bin));
 };
 
+let info_gen_identity = {
+  let doc = "Create a public-private key pair for a validator";
+  Term.info(
+    "generate-identity",
+    ~version="%‌%VERSION%%",
+    ~doc,
+    ~exits,
+    ~man,
+  );
+};
+
+let self_uri = {
+  let docv = "self_uri";
+  let doc = "The uri that other nodes should use to connect to this node.";
+  Arg.(required & opt(some(uri), None) & info(["uri"], ~doc, ~docv));
+};
+
+let gen_identity = {
+  let f = uri => {
+    open Tezos_interop;
+    let (sk, pk) = Mirage_crypto_ec.Ed25519.generate();
+    module Identity = {
+      [@deriving yojson]
+      type t = {
+        secret_key: string,
+        public_key: string,
+        uri: Uri.t,
```

Comment:
why not use the [files api](https://github.com/marigold-dev/sidechain/blob/78866292ce5da54ed21d47483123e884adc8216c/bin/files.re#L19-L22) here?

Reply:
Because these are only meant to logged to the console - the way @EduardoRFS wanted it.